src/contexts/Transaction/WagmiTxHistoryProvider.tsx

```diff
@@ -1,14 +1,94 @@
-import { ReactNode } from 'react'
-import { TxHistoryContext } from './TxHistoryContext'
+import { getTransactionReceipt } from '@wagmi/core'
+import { TxStatus } from 'models/transaction'
+import { wagmiConfig } from 'packages/v4/wagmiConfig'
+import { ReactNode, useEffect } from 'react'
+import { TransactionLog, TxHistoryContext } from './TxHistoryContext'
 import { useTransactions } from './useTransactions'
 
+const nowSeconds = () => Math.round(new Date().valueOf() / 1000)
+
+const SHORT_POLL_INTERVAL_MILLISECONDS = 3 * 1000 // 3 sec
+const LONG_POLL_INTERVAL_MILLISECONDS = 12 * 1000 // 12 sec
+
+const pollTransaction = async (
+  txLog: TransactionLog,
+): Promise<TransactionLog> => {
+  // Only do refresh logic for pending txs
+  // tx.hash shouldn't ever be undefined but it's optional typed :shrug:
+  if (!txLog.tx?.hash) {
+    return txLog
+  }
+
+  try {
+    // this throws if the tx hasnt been mined
+    const response = await getTransactionReceipt(wagmiConfig, {
+      hash: txLog.tx.hash,
+    })
+
+    return {
+      ...txLog,
+      status:
+        response.status === 'success' ? TxStatus.success : TxStatus.failed,
+    }
+  } catch {
+    return txLog
+  }
+
+  return txLog
+}
+
 export default function WagmiTxHistoryProvider({
   children,
 }: {
   children: ReactNode
 }) {
-  const { transactions, addTransaction, removeTransaction } = useTransactions()
-  // TODO implement polling/wait logic using Wagmi
+  const { transactions, addTransaction, removeTransaction, setTransactions } =
```

Comment:
This all ends up doing the exact same thing as the Ethers variant, except the `pollTransaction` function is wagmi-specific.

I tried to improve it but meh.